### PR TITLE
docs: state variable

### LINKED ISSUE #65: Add Documentation on State Logic (or make sure it already exists)
Add a documentation page.

### PR DESCRIPTION
**Pull request by hollandjg**
# Description

Add really basic introductions to the State and Variable objects, and their fundamental properties.

This is meant to be complementary to the cycle tutorials which go into more in-depth examples. 

*If the issue is on Github, simply link to it using the '#' symbol; otherwise, provide a notion page link)*:
- resolves #65 

# Type of change

- **docs**: Documentation only changes

# Features (Optional)
- Just added documentation pages

# Questions (Optional)
- Are these understandable?

### CODE REVIEW

File: docs/The State Mechanism.ipynb [hunk 1]
```diff
@@ -0,0 +1,618 @@
...
+    "| Procedure         | Data            |\n",
+    "|-------------------|-----------------|\n",
+    "| Experimentalist   | Conditions      |\n",
+    "| Experiment Runner | Experiment Data |\n",
+    "| Theorist          | Model           |\n",
+    "\n",
+    "The data produced by each procedure $f$ can be seen as additions to the existing data. Each procedure $f$:\n",
+    "- Takes in existing Data in a `State` $S$\n",
+    "- Adds new data $\\Delta S$\n",
+    "- Returns an updated state of the Data $S^\\prime$  \n",
```
Comment (benwandrew) [inline docs/The State Mechanism.ipynb:43]:
```suggestion
    "- Returns an updated `State` $S^\\prime$  \n",
```

File: docs/The State Mechanism.ipynb [hunk 2]
```diff
@@ -0,0 +1,618 @@
...
+       "0  1  1)"
+      ]
+     },
+     "execution_count": null,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "s + autora.state.Delta(data=pd.DataFrame({\"x\":[1], \"y\":[1]}))"
```
Comment (benwandrew) [inline docs/The State Mechanism.ipynb:178]: is this supposed to be s_0 = ... or no?

File: docs/Variable.ipynb [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f4ab2b25903f40a7",
+   "metadata": {},
+   "source": [
+    "The `autora.state.estimator_from_state` function uses the `names` of the variables to pass the correct columns to a \n",
+    "scikit-learn compatible estimator for curve fitting."
```
Comment (benwandrew) [inline docs/Variable.ipynb:295]:
```suggestion
    "`scikit-learn` compatible estimator for curve fitting."
```

**Review by benwandrew — CHANGES_REQUESTED** [review]

File: docs/The State Mechanism.ipynb (hunk 2 above)
Comment (hollandjg) [reply docs/The State Mechanism.ipynb:178]: No, the text which follows it is wrong. I've added an update. Does this make more sense?

Comment (benwandrew) [reply docs/The State Mechanism.ipynb:178]: yes, it does!

**Review by benwandrew — APPROVED** [review]
tests are taking a while, but looks good to me!

**Pull request merged**